### PR TITLE
DB/ Created Airdrop table in DB, Added crud methods

### PR DESCRIPTION
**Pull request by binayak9932**
issue #104

### CODE REVIEW

File: web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py [hunk 1]
```diff
@@ -30,13 +30,15 @@ def upgrade() -> None:
     if column_exists('position', 'start_price'):
         print("Column 'start_price' already exists, skipping creation.")
     else:
-        op.add_column('position', sa.Column('start_price', sa.DECIMAL(), nullable=False))
+        op.add_column('position', sa.Column('start_price', sa.DECIMAL(), nullable=False,
```
Comment (djeck1432) [inline web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py:33]: Why did you change old migration?

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    Additional configuration:
+    
+    - Foreign key constraint on `user_id` to reference the `user` table.
+    - Primary key constraint on `id`.
+    - Index on `user_id` for optimized querying based on the `user_id` field.
+
+    This function is part of the Alembic migration and is auto-generated.
+    Adjustments may be made if additional configuration or constraints are needed.
+    """
+    op.create_table('airdrop',
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:42]: Please, handle case if this table already created

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+def downgrade() -> None:
+    """
+    Perform the downgrade migration to remove the 'airdrop' table from the database.
+
+    This migration drops the 'airdrop' table and its associated index on `user_id`. 
+    It is intended to reverse the changes made in the `upgrade` function, allowing 
+    for a rollback of the database schema to the state before the 'airdrop' table was added.
+    """
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index(op.f('ix_airdrop_user_id'), table_name='airdrop')
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:65]: Handle case if this table already dropped

File: web_app/db/crud.py [hunk 4]
```diff
@@ -360,4 +376,38 @@ def get_total_amounts_for_open_positions(self) -> float | None:
...
+            logger.error(f"AirDrop with ID {airdrop_id} not found")
+
+    def get_all_unclaimed(self) -> List[AirDrop]:
+        """
+        Returns all AirDrop instances where is_claimed=False.
+        :return: List[AirDrop]
+        """
+        with self.Session() as db:
+            try:
+                unclaimed = db.query(AirDrop).filter(AirDrop.is_claimed == False).all()
```
Comment (djeck1432) [inline web_app/db/crud.py:409]: please, be more specific, unclaimed what? I'd do like `unclaimed_records` or `unclaimed_instances` 
use `filter_by` instead `filter`

File: web_app/db/models.py [hunk 5]
```diff
@@ -73,3 +73,20 @@ class Position(Base):
...
+
+    __tablename__ = "airdrop"
+
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
+    user_id = Column(
+        UUID(as_uuid=True), ForeignKey("user.id"), index=True, nullable=False
+    )
+    created_at = Column(DateTime, nullable=False, default=func.now())
+    amount = Column(DECIMAL, nullable=True)
+    is_claimed = Column(Boolean, default=False)
```
Comment (djeck1432) [inline web_app/db/models.py:91]: add index to `is_claimed` as we have query where we are filtering by this field

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py (hunk 1 above)
Comment (binayak9932) [reply web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py:33]: without this config  its giving error

**Comment by binayak9932** [comment]
@djeck1432  please review

File: web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py (hunk 1 above)
Comment (djeck1432) [reply web_app/alembic/versions/a009512f5362_start_price_field_added_in_position.py:33]: which error, please ,be more specific, if you have any error, attach it here with explanation what did you run the command to get this error and traceback of this error

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            sa.ForeignKeyConstraint(
+                ["user_id"],
+                ["user.id"],
+            ),
+            sa.PrimaryKeyConstraint("id"),
+        )
+        op.create_index(
+            op.f("ix_airdrop_user_id"), "airdrop", ["user_id"], unique=False
+        )
+        print("Table 'airdrop' created successfully.")
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:67]: remove `print` use `logging`

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    else:
+        print("Table 'airdrop' already exists, skipping creation.")
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:69]: remove `print` use `logging`

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    This migration drops the 'airdrop' table and its associated index on `user_id`.
+    It is intended to reverse the changes made in the `upgrade` function, allowing
+    for a rollback of the database schema to the state before the 'airdrop' table was added.
+    """
+    bind = op.get_bind()
+    inspector = Inspector.from_engine(bind)
+    if "airdrop" in inspector.get_table_names():
+        op.drop_index(op.f("ix_airdrop_user_id"), table_name="airdrop")
+        op.drop_table("airdrop")
+        print("Table 'airdrop' and its index were dropped successfully.")
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:84]: remove `print` use `logging`

File: web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    else:
+        print("Table 'airdrop' does not exist, skipping drop.")
```
Comment (djeck1432) [inline web_app/alembic/versions/e69320e12cc7_add_airdrop_model.py:86]: remove `print` use `logging`

File: web_app/db/models.py [hunk 10]
```diff
@@ -73,3 +73,20 @@ class Position(Base):
...
+
+    __tablename__ = "airdrop"
+
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
+    user_id = Column(
+        UUID(as_uuid=True), ForeignKey("user.id"), index=True, nullable=False
+    )
+    created_at = Column(DateTime, nullable=False, default=func.now())
+    amount = Column(DECIMAL, nullable=True)
+    is_claimed = Column(Boolean, default=False, index=True)
```
Comment (djeck1432) [inline web_app/db/models.py:91]: please, add this index also to migration, I wrote it here, maybe you forgot to regenerate migrations

File: web_app/tests/test_dbconnector.py [hunk 11]
```diff
@@ -100,3 +100,46 @@ def test_delete_object_invalid_id(mock_db_connector):
     """
     non_existent_id = uuid.uuid4()
     mock_db_connector.delete_object(non_existent_id)
+
```
Comment (djeck1432) [inline web_app/tests/test_dbconnector.py:103]: Remove unnecessary blank or run `black` command

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by binayak9932** [comment]
@djeck1432  please  review

**Review by djeck1432 — APPROVED** [review]
Looks good to me, approved
